samples/webrequest/webrequest.cpp

```diff
@@ -0,0 +1,413 @@
+
+        // Prepare UI controls
+        wxSizer* mainSizer = new wxBoxSizer(wxVERTICAL);
+
+        mainSizer->Add(new wxStaticText(this, wxID_ANY, "Request URL:"),
+            wxSizerFlags().Border());
+        m_urlTextCtrl = new wxTextCtrl(this, wxID_ANY,
+            "https://www.wxwidgets.org/downloads/logos/blocks.png");
+        mainSizer->Add(m_urlTextCtrl,
+            wxSizerFlags().Expand().Border(wxLEFT | wxRIGHT));
+
+        m_notebook = new wxNotebook(this, wxID_ANY);
+        m_notebook->Bind(wxEVT_NOTEBOOK_PAGE_CHANGED, &WebRequestFrame::OnNotebookPageChanged, this);
+
+        // Image page
+        wxPanel* imagePanel = new wxPanel(m_notebook);
+        wxSizer* imageSizer = new wxBoxSizer(wxVERTICAL);
+
+        m_imageStaticBitmap = new wxStaticBitmap(imagePanel,
+            wxID_ANY, wxArtProvider::GetBitmap(wxART_MISSING_IMAGE));
+        imageSizer->Add(m_imageStaticBitmap, wxSizerFlags(1).Expand());
+
+        imagePanel->SetSizer(imageSizer);
+        m_notebook->AddPage(imagePanel, "Image", true);
+
+        // Text page
+        wxPanel* textPanel = new wxPanel(m_notebook);
+        wxSizer* textSizer = new wxBoxSizer(wxVERTICAL);
+
+        m_postCheckBox = new wxCheckBox(textPanel, wxID_ANY, "Post request body");
+        textSizer->Add(m_postCheckBox, wxSizerFlags().Border());
+        m_postCheckBox->Bind(wxEVT_CHECKBOX, &WebRequestFrame::OnPostCheckBox, this);
+
+        m_postRequestTextCtrl = new wxTextCtrl(textPanel, wxID_ANY,
+            "app=WebRequestSample&version=1",
+            wxDefaultPosition, wxSize(-1, FromDIP(60)), wxTE_MULTILINE);
+        textSizer->Add(m_postRequestTextCtrl,
+            wxSizerFlags().Expand().Border(wxLEFT | wxRIGHT));
+
+        textSizer->Add(new wxStaticText(textPanel, wxID_ANY, "Request body content type:"),
+            wxSizerFlags().Border());
+        m_postContentTypeTextCtrl = new wxTextCtrl(textPanel, wxID_ANY,
+            "application/x-www-form-urlencoded");
+        textSizer->Add(m_postContentTypeTextCtrl,
+            wxSizerFlags().Expand().Border(wxLEFT | wxRIGHT));
+
+        textSizer->Add(new wxStaticText(textPanel, wxID_ANY, "Response body:"),
+            wxSizerFlags().Border());
+        m_textResponseTextCtrl = new wxTextCtrl(textPanel, wxID_ANY, "",
+            wxDefaultPosition, wxDefaultSize, wxTE_MULTILINE | wxTE_READONLY);
+        m_textResponseTextCtrl->SetBackgroundColour(wxSystemSettings::GetColour(wxSYS_COLOUR_BTNFACE));
+        textSizer->Add(m_textResponseTextCtrl,
+            wxSizerFlags(1).Expand().Border(wxLEFT | wxRIGHT | wxBOTTOM));
+
+        textPanel->SetSizer(textSizer);
+        m_notebook->AddPage(textPanel, "Text");
+
+        // Download page
+        wxPanel* downloadPanel = new wxPanel(m_notebook);
+        wxSizer* downloadSizer = new wxBoxSizer(wxVERTICAL);
+        wxStaticText* downloadHeader = new wxStaticText(downloadPanel, wxID_ANY,
+            "The URL will be downloaded to a file.\n"\
```

Comment:
```suggestion
            "The URL will be downloaded to a file.\n"
```
Escaping the newline is not needed outside the quotes. There are several cases like this below.